mantidproject/mantid
# Fixes for sliceviewer line plots

**Description of work.**

Several fixes to address problems with the line plots in sliceviewer:
* avoiding an error with setting normalization options too early
* disables autoscaling of lineplots on axis linked to image. This avoids the image plot immediately zooming out when the mouse moves over the image.
* fixes the home button to simply reset the data limits rather than use the matplotlib navstack

**To test:**

# Simple test of lines
```python
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
nrows, ncols = 5, 5
signal = np.arange(0,nrows*ncols)
x = np.arange(0,2*ncols+1,step=2)
ws = CreateWorkspace(DataX=x, DataY=signal, NSpec=nrows,UnitX='TOF')
```
creates a simple workspace with increasing values. Check this plots okay, the values displayed in the cursor coordinates are correct and the line plots make sense as the mouse moves around.

# Test with MD workspace

The files are in the [training data](http://sourceforge.net/projects/mantid/files/Sample%20Data/TrainingCourseData.zip/download)
```
SXD23767 = Load(Filename='SXD23767.raw', LoadMonitors='Exclude')
mdws = ConvertToMD(
    InputWorkspace=SXD23767,
    QDimensions="Q3D",
    dEAnalysisMode="Elastic",
    QConversionScales="Q in A^-1",
    LorentzCorrection='1',
    MinValues=[-15, -15, -15],
    MaxValues=[15, 15, 15],
    SplitInto='2',
    SplitThreshold='50',
    MaxRecursionDepth='14')
peaksws = Load('peaks_qLab.nxs')
```
* show the sliceviewer on `mdws`
* click the peaks overlay and select `peaksws`
* in the table click on the row `123` that should zoom to a peak.
* activate the line plots and move the mouse around over the squares. The peaks in the lines should be in the center of the pixel.

Refs #28579.

*This does not require release notes* because **it fixes a regression**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
